quic/packets.nim

```diff
@@ -9,43 +9,69 @@ export length
 
 {.push raises:[].} # avoid exceptions in this module
```

Comment:
We should probably still allow for `Defect`?

Reply:
I tried using `{.push raises:[Defect].}`, but that led to a bunch of compiler warnings.

Reply:
We typically set to Defect as not all Defects are properly tracked currently, so to be sure. 
Have to live with the warnings indeed.. :(

Reply:
Thanks for the pointers. Please bear with me as I try to fully understand this :) I've been reading some issues in the Nim repo: 
  - 11776 - "Exception should not be Defect"
  - 12862 - "The Defect exceptions are not consistently tracked"
  - RFC 77 - "Exceptions that inherit from system.Defect should not be tracked". 

From these issues I conclude that in newer versions of Nim, `Defect` is not tracked anymore but in older versions of Nim it is tracked inconsistently.

Would I be correct in assuming that because there are no callbacks in our code, and because the compiler is not complaining about `Defect`, that the `{.push raises:[].}` is ok for now? If the need arises, we can always add `Defect`, but it means we get to keep the compilation warning free for now.

Reply:
there are bugs in the compiler where things that raise defect don't cause a corresponding `Defect` in the tracking system - for example `+` raises overflow which is a defect, but the exception tracking mechanism is broken / buggy and the effect doesn't show up - we thus tend to use `raises: [Defect]` even if it causes a warning because the code most likely de facto does raise a defect

Reply:
From what I read in RFC 77, it seems intentional that overflow and other defects do not show up in the exception tracking mechanism (for newer versions of Nim).

Reply:
I included Defect in the list of exceptions, after our offline discussion yesterday. We concluded that it's better to include it because Nim handles Defect inconsistently. See commit d1d083d.